agent/proxycfg/snapshot.go

```diff
@@ -124,28 +168,38 @@ type ConfigSnapshot struct {
 
 	ServerSNIFn ServerSNIFunc
 	Roots       *structs.IndexedCARoots
+	// Used for connect-proxy and ingress-gateway proxies
+	Leaf *structs.IssuedCert
```

Comment:
It's a little weird to see this promoted up to a shared field when it's not used by the mesh-gateway. Having said that, I think it's already true for `Roots` so it's probably fine.

Reply:
Note: Mesh Gateways have to watch the roots to get the trust domain.

Reply:
I completely forgot that detail.

Reply:
Moved this into a shared data structure not at the top-level